### PR TITLE
Remove evaluation loop legacy dict returns for `*_epoch_end` hooks

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

Split out from #6969

This seems like it's been deprecated since v0.9.1 and doesn't change the behavior.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/trainer/evaluation_loop.py [hunk 1]
```diff
@@ -231,13 +231,6 @@ def __run_eval_epoch_end(self, num_dataloaders):
...
         self.trainer.logger_connector.cache_logged_metrics()
-        # depre warning
-        if eval_results is not None and user_reduced:
-            step = 'testing_epoch_end' if self.trainer.testing else 'validation_epoch_end'
-            self.warning_cache.warn(
-                f'The {step} should not return anything as of 9.1.'
-                ' To log, use self.log(...) or self.write(...) directly in the LightningModule'
-            )
 
         if not isinstance(eval_results, list):
```
Comment (carmocca) [inline pytorch_lightning/trainer/evaluation_loop.py:235]: Don't you want to also remove this?

**Review by carmocca — COMMENTED** [review]
Looks like we had a test checking this warning message

File: pytorch_lightning/trainer/evaluation_loop.py (hunk 1 above)
Comment (ananthsub) [reply pytorch_lightning/trainer/evaluation_loop.py:235]: yeah will do

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — COMMENTED** [review]
thanks for the additional cleanup @carmocca !

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by kaushikb11 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM ! Thanks.

**Pull request merged**